src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -70,9 +70,37 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           r.resolve(packageRelativeUrl`http://abc.xyz/foo.html`),
           resolvedUrl`http://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`https://baz.com/qux.js`,
+              fileRelativeUrl`https://foo.com/bar.html`),
+          resolvedUrl`https://foo.com/bar.html`);
+    });
+
+    test('resolves protocol-relative URLs using default protocol', () => {
+      const r = new PackageUrlResolver();
       assert.equal(
           r.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
-          resolvedUrl`file://abc.xyz/foo.html`);
+          resolvedUrl`https://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`http://foo.com/bar.html`,
+              fileRelativeUrl`//foo.com/bar.html`),
+          resolvedUrl`https://foo.com/bar.html`);
+    });
+    test('resolves protocol-relative URLs using provided protocol', () => {
+      const r = new PackageUrlResolver({protocol: 'potato:'});
+      assert.equal(
+          r.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
+          resolvedUrl`potato://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`https://foo.com/bar.html`,
+              fileRelativeUrl`//foo.com/bar.html`),
+          resolvedUrl`potato://foo.com/bar.html`);
```

Comment:
If I follow what you're saying, there's a specific exclusion around inheriting file-protocol from base URL but otherwise we should inherit protocol from a base URL (when provided) like...
```ts
      assert.equal(
          r.resolve(packageRelativeUrl`//foo.com/bar.html`),
          rootedFileUrl`1/2/bar.html`,
          'host matches resolver, resolve to file URL');

      assert.equal(
          r.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
          resolvedUrl`potato://abc.xyz/foo.html`,
          'host does not match resolver, resolve with protocol option');

      assert.equal(
          r.resolve(
              resolvedUrl`spaghetti://abc.xyz/foo.html`,
              fileRelativeUrl`//foo.com/bar.html`),
          rootedFileUrl`1/2/bar.html`,
          'host matches resolver, resolve as file URL');

      assert.equal(
          r.resolve(
              resolvedUrl`spaghetti://abc.xyz/foo.html`,
              fileRelativeUrl`//abc.xyz/foo.html`),
          resolvedUrl`spaghetti://abc.xyz/foo.html`,
          'host does not match resolver, inherit non-file protocol from base URL');

      assert.equal(
          r.resolve(
              rootedFileUrl`1/2/bar.html`, fileRelativeUrl`//abc.xyz/foo.html`),
          resolvedUrl`potato://abc.xyz/foo.html`,
          'host does not match resolver, do not inherit file protocol from base URL');

      assert.equal(
          r.resolve(
              resolvedUrl`spaghetti://example.com/`,
              fileRelativeUrl`//abc.xyz/foo.html`),
          resolvedUrl`spaghetti://abc.xyz/foo.html`,
          'host does not match resolver, inherit non-file protocol from base URL');
```

Reply:
Big +1 to these assertions, I think this is our intended behavior.